src/orquestra/sdk/_base/_api/_wf_run.py

```diff
@@ -605,6 +611,63 @@ def _handle_common_listing_project_errors(
     return None
 
 
+def _ask_runtimes_for_config(
+    wf_run_id: WorkflowRunId, project_dir: Path
+) -> RuntimeConfig:
+    not_found_configs: t.List[RuntimeConfig] = []
+    unauthorized_configs: t.List[RuntimeConfig] = []
+    not_running_configs: t.List[RuntimeConfig] = []
+
+    config_names = RuntimeConfig.list_configs()
+    for config_name in config_names:
+        config_obj = RuntimeConfig.load(config_name)
+        try:
+            runtime = config_obj._get_runtime(project_dir)
+        except RayNotRunningError:
+            # Ray connection is set up in `RayRuntime.__init__()`. We'll get the
+            # exception when runtime object is created.
+            not_running_configs.append(config_obj)
+            continue
+
+        try:
+            _ = runtime.get_workflow_run_status(wf_run_id)
+        except WorkflowRunNotFoundError:
+            not_found_configs.append(config_obj)
+            continue
+        except UnauthorizedError:
+            # TODO (ORQSDK-990): short circuit remote call by checking token validity
+            # on the client side
+            unauthorized_configs.append(config_obj)
+            continue
+
+        # We're here = the runtime knows about this run ID.
+        return config_obj
+
+    raise RuntimeQuerySummaryError(
+        wf_run_id=wf_run_id,
+        not_found_runtimes=[_make_runtime_info(config) for config in not_found_configs],
+        unauthorized_runtimes=[
+            _make_runtime_info(config) for config in unauthorized_configs
+        ],
+        not_running_runtimes=[
+            _make_runtime_info(config) for config in not_running_configs
+        ],
+    )
+
+
+def _make_runtime_info(config: RuntimeConfig) -> RuntimeQuerySummaryError.RuntimeInfo:
+    # TODO: add public properties to config
```

Comment:
Issue: Unfinished TODO. Either remove TODO, actually implement it or create a link and add it here in comment